### PR TITLE
Fix namespace interceptor to allow search attributes operations

### PR DESCRIPTION
**Pull request by rodrigozhou**
**What changed?**
Let search attributes operations requests to pass without namespace set in the namespace interceptor.

**Why?**
Namespace interceptor validates the namespace argument of all requests if present.
However, this argument is optional for search attributes operations.

**How did you test it?**
Start server, and run tctl operations for search attributes no longer returns error "Namespace not set on request".

**Potential risks**
None.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: proto/internal/temporal/server/api/adminservice/v1/request_response.proto [hunk 1]
```diff
@@ -209,15 +209,15 @@ message AddSearchAttributesResponse {
...
+    string namespace = 3;
 }
 
 message RemoveSearchAttributesResponse {
 }
 
 message GetSearchAttributesRequest {
     string index_name = 1;
-    string namespace = 4;
+    string namespace = 2;
```
Comment (rodrigozhou) [inline proto/internal/temporal/server/api/adminservice/v1/request_response.proto:220]: This is just a fix in the arguments numbers. These are new arguments, and were not updated in tctl or cli yet. It should be safe.

File: common/rpc/interceptor/namespace_validator.go [hunk 2]
```diff
@@ -178,6 +180,18 @@ func (ni *NamespaceValidatorInterceptor) extractNamespaceFromRequest(req interfa
...
 		return nil, nil
+	case *adminservice.AddSearchAttributesRequest,
+		*adminservice.RemoveSearchAttributesRequest,
+		*adminservice.GetSearchAttributesRequest,
+		*operatorservice.AddSearchAttributesRequest,
+		*operatorservice.RemoveSearchAttributesRequest,
+		*operatorservice.ListSearchAttributesRequest:
+		// Namespace is optional for search attributes operations.
+		// It's required when using SQL DB for visibility, but not when using Elasticsearch.
+		if !namespaceName.IsEmpty() {
```
Comment (Spikhalskiy) [inline common/rpc/interceptor/namespace_validator.go:191]: 👍

**Review by Spikhalskiy — APPROVED** [review]

**Comment by Spikhalskiy** [comment]
Should here be a unit test fixing the behavior?

**Review by mindaugasrukas — APPROVED** [review]
LGTM

**Comment by rodrigozhou** [comment]
@Spikhalskiy Added unit tests.

**Pull request merged**